inpage/src/Button.tsx

```diff
@@ -39,24 +42,76 @@ const sheet = jss.createStyleSheet({
       background: color(fgColor).alpha(0.15).toString(),
     },
   },
+  ButtonDisabled: {
+    filter: 'brightness(50%)',
+    cursor: 'initial',
+  },
 });
 
 sheetsRegistry.add(sheet);
 
 const Button = ({
   children,
   secondary,
+  disabled,
+  onPress = () => undefined,
   ...props
-}: Omit<HTMLProps<HTMLDivElement>, 'className'> & { secondary?: boolean }) => (
-  <div
-    {...props}
-    {...classes(
-      sheet.classes.Button,
-      secondary && sheet.classes.ButtonSecondary,
-    )}
-  >
-    {children}
-  </div>
-);
+}: Omit<HTMLProps<HTMLDivElement>, 'className' | 'onClick'> & {
+  secondary?: boolean;
+  onPress?: (
+    e: React.MouseEvent<HTMLDivElement> | React.KeyboardEvent<HTMLDivElement>,
+  ) => unknown;
+}) => {
+  const [loading, setLoading] = useState(false);
+  const [_error, setError] = useState<unknown>();
+
+  const handlePress = useCallback(
+    (
+      e: React.MouseEvent<HTMLDivElement> | React.KeyboardEvent<HTMLDivElement>,
+    ) => {
+      if (disabled || loading) {
+        return;
+      }
+
+      (async () => {
+        try {
+          setLoading(true);
+          await onPress(e);
+          setError(undefined);
+        } catch (newError) {
+          // eslint-disable-next-line no-console
+          console.error(newError);
+          setError(newError);
+        }
+
+        setLoading(false);
+      })().catch(() => assert(false));
```

Comment:
What is the purpose of `assert(false)`?

Reply:
If the async function throws an exception, then the `(async () => { ... })()` expression simply evaluates to a rejected promise. This is something that eslint is picking up. Adding `.catch(() => assert(false))` throws an exception.

It's probably not ideal though, since modern JS runtimes will report uncaught exceptions in promises anyway. I'll take a look at this and maybe just disable the rule.